src/plugins/discover/public/application/components/table/table.test.tsx

```diff
@@ -404,15 +403,32 @@ describe('DocViewTable at Discover Doc with Fields API', () => {
     expect(findTestSubject(component, 'tableDocViewRow-customer_first_name.nickname').length).toBe(
       1
     );
+
+    expect(
+      findTestSubject(component, 'tableDocViewRow-category.keyword-multifieldBadge').length
+    ).toBe(1);
+
+    expect(
+      findTestSubject(component, 'tableDocViewRow-customer_first_name.keyword-multifieldBadge')
+        .length
+    ).toBe(1);
+
+    expect(
+      findTestSubject(component, 'tableDocViewRow-customer_first_name.nickname-multifieldBadge')
+        .length
+    ).toBe(1);
   });
 
   it('does not render multifield rows if showMultiFields flag is not set', () => {
-    const component = mount(<DocViewTable {...props} />);
-    const categoryMultifieldRow = findTestSubject(
-      component,
-      'tableDocViewRow-multifieldsTitle-category'
-    );
-    expect(categoryMultifieldRow.length).toBe(0);
+    (getServices as jest.Mock).mockImplementationOnce(() => ({
+      uiSettings: {
+        get: (key: string) => {
+          return key === 'discover:showMultiFields' && false;
+        },
+      },
+    }));
+    const component = mountComponent(props);
+
     const categoryKeywordRow = findTestSubject(component, 'tableDocViewRow-category.keyword');
```

Comment:
Since the implementation has changed a bit, should be testing for the lack of `multifieldBadge` here, same as in the previous test?

Reply:
Good point! Added.